maas/resource_maas_boot_source.go

```diff
@@ -0,0 +1,30 @@
+package maas
+
+import (
+	"fmt"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+)
+
+func findBootSource(client *client.Client) (*entity.BootSource, error) {
+	bootsources, err := client.BootSources.Get()
+	if err != nil {
+		return nil, err
+	}
+	if len(bootsources) > 0 {
+		return &bootsources[0], nil
+	}
+	return nil, nil
+}
+
+func getBootSource(client *client.Client) (*entity.BootSource, error) {
+	bootsource, err := findBootSource(client)
```

Comment:
what if we simplify it a bit? Let's remove the second function findBootSource, return always the first bootsource found and also fail if the length is different than 1? The error is something like "expected a single boot source" or "cannot support more than one", or something similar?